contracts/account/extensions/AccountERC6900.sol

```diff
@@ -0,0 +1,494 @@
+            }
+        }
+        return
+            IERC6900ValidationModule(module).validateUserOp(entityId, userOp, _signableUserOpHash(userOp, userOpHash));
+    }
+
+    /**
+     * @dev Lowest-level signature validation function. See {ERC7739-_rawSignatureValidation}.
+     *
+     * This function delegates the signature validation to a validation module if the first 20 bytes of the
+     * signature correspond to an installed validator module.
+     *
+     * See {_extractSignatureValidator} for the module extraction logic.
+     */
+    function _rawSignatureValidation(
+        bytes32 hash,
+        bytes calldata signature
+    ) internal view virtual override returns (bool) {
+        if (signature.length < 25) return false;
+        (ModuleEntity validationModuleEntity, bytes calldata innerSignature) = _extractSignatureValidator(signature);
+        (address module, uint32 entityId) = validationModuleEntity.unpack();
+        // If the validation function is attempted to be used for signature validation
+        // and the flag isSignatureValidation is set to false, validation MUST revert
+        require(
+            !_validations[validationModuleEntity].validationFlags.isSignatureValidation(),
+            ERC6900Utils.ERC6900BadSignatureValidation(validationModuleEntity)
+        );
+
+        return
+            IERC6900ValidationModule(module).validateSignature(
+                address(this),
+                entityId,
+                msg.sender,
+                hash,
+                innerSignature
+            ) == IERC1271.isValidSignature.selector;
+    }
+
+    /**
+     * @dev ERC-6900 execution logic.
+     *
+     * Reverts if the call type is not supported.
+     */
+    function _execute(address target, uint256 value, bytes calldata data) internal virtual returns (bytes memory) {
+        (bool success, bytes memory returndata) = target.call{value: value}(data);
+
+        require(success, ERC6900Utils.ERC6900ExecutionFailed(target, value, data));
+        return returndata;
+    }
+
+    /**
+     * @dev Installs a validation module of the given type with the given initialization data.
+     *
+     *
+     * Requirements:
+     *
+     * * TODO
+     *
+     * Emits a {ValidationInstalled} event.
+     */
+    function _installValidation(
+        ValidationConfig validationConfig,
+        bytes4[] calldata selectors,
+        bytes calldata installData,
+        bytes[] calldata hooks
+    ) internal virtual {
+        (ModuleEntity moduleEntity, ValidationFlags _validationFlags) = validationConfig.unpack();
+        (address module, uint32 entityId) = moduleEntity.unpack();
+        // Modules MUST implement ERC-165 for IModule.
+        require(
+            ERC165Checker.supportsInterface(module, type(IERC6900Module).interfaceId),
+            ERC6900Utils.ERC6900ModuleInterfaceNotSupported(module, type(IERC6900Module).interfaceId)
+        );
```

Comment:
We can't always call `supportsInterface` since the module may be an EOA (or other non-module) used for direct validation.

Reply:
I'm not sure to understand, can a module be an EOA?

Reply:
See the paragraph on [direct call validation](https://eips.ethereum.org/EIPS/eip-6900#direct-call-validation) in the ERC. Yes, you can install a validation function with the module being an EOA.

Reply:
Hey all, I wanted to mention here that the ERC-6900 [spec](https://eips.ethereum.org/EIPS/eip-6900) does not require that this check happens onchain, and we realized that having this as non-standard behavior in the reference implementation is probably confusing. So, we've decided to remove it from the RI here: https://github.com/erc6900/reference-implementation/pull/214

Per our discussion with @Amxx and @ernestognw, we're still requiring this to be implemented by modules, and moving the responsibility of checking to be off-chain (i.e. by wallets) prior to installing a module.